packages/wrangler/src/dev/miniflare.ts

```diff
@@ -867,6 +870,41 @@ export async function buildMiniflareOptions(
 
 		...persistOptions,
 		workers: [
+			...getAssetServerWorker(config),
+			...userWorkerOptions,
+			...externalWorkers,
+		],
+	};
+	return { options, internalObjects, entrypointNames };
+}
+
+async function getUserWorker(config: Omit<ConfigBundle, "rules">): Promise<{
+	userWorkerOptions: WorkerOptions[];
+	entrypointNames: string[];
+	internalObjects: CfDurableObject[];
+	externalWorkers: WorkerOptions[];
+}> {
+	const { bindingOptions, internalObjects, externalWorkers } =
+		buildMiniflareBindingOptions(config);
+	// if static assets only, no user worker:
+	if (
+		config.experimentalAssets &&
+		config.bundle.entry.file ===
+			path.resolve(getBasePath(), "templates/no-op-worker.js")
```

Comment:
Is this PR only a temporary step?
If not, why doe we still have a `bundle.entry` at all if  it is assets-only?
It looks like we are still creating this entry but marking it with the `no-op-worker.js` to indicate it is not a real one?

Reply:
my concern was that not having an entry worker would require us to put all sorts of `if..else`s in place to by-pass the different layers of bundling we currently have in place. Felt rather risky too, since we'd need to track down all layers where bundling is happening, and was worried we might break existing things, introduce too much complexity, etc. So my suggestion was to use a no-op worker that we can just ignor in assets-only mode.

Reply:
Yes I originally wanted to never use the no-op worker as an entry at all, but the problem is *a lot* of wrangler assumes there is a user worker. Talked with Carmen and we thought it might be better to keep the changes for this more localised and just remove the no-op worker at the last minute. 

But I'm very open to other solutions, or if you think its fine to let the entry/main module be undefined and add in the appropriate checks around the codebase, then I'm happy to do that :)